lib/resty/waf/request.lua

```diff
@@ -133,6 +133,15 @@ function _M.parse_request_body(waf, request_headers, collections)
 		-- remove charset from the content-type (e.g. application/json;charset=utf-8 -> application/json)
 		content_type_header = string.match(content_type_header, "[^;]+")
 		
+		-- check that the request has a content length, which otherwise triggers an error with HTTP2 and HTTP3 requests
+		-- https://github.com/openresty/lua-nginx-module/commit/86bea01b244938e0ab6eda780906c06fa62c3b5c
+		if ngx.req.http_version() >= 2 and request_headers["content-length"] == nil then
+			logger.warn(waf, "HTTP2/HTTP3 request without a Content-Length header")
+			if waf._mode == "ACTIVE" then
+				ngx.exit(ngx.HTTP_FORBIDDEN)
```

Comment:
I'd suggest we return a 400 error instead of 403 as this is an RFC-compliance issue, like other errors above.

```suggestion
				ngx.exit(400)
```

Reply:
Sure, good idea